connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -573,9 +582,11 @@ public Void call() throws Exception {
                     return null;
                 }
 
-                if (worker.ownsConnector(connName)) {
+                if (assignment.connectors().contains(connName)) {
```

Comment:
I guess the reason for this check is that we catch exceptions starting the connector from the herder. I'm wondering whether you considered catching them in the worker instead?